cmd/root.go

```diff
@@ -185,9 +185,11 @@ func outputJSON(results []checker.CheckResult) {
 
 	for _, r := range results {
 		tmpResult := checker.CheckResult{
-			Name:       r.Name,
-			Pass:       r.Pass,
-			Confidence: r.Confidence,
+			Name:        r.Name,
+			Pass:        r.Pass,
+			Confidence:  r.Confidence,
+			Description: r.Description,
```

Comment:
I also don't think we should include these long strings in these results by default. It will blow up the latest.json and cause a lot of duplication. 

If we need this for another use case, we should figure out another way to do this.

Reply:
I can think of a flag that either dumps the help information in the json or not. 

Do you have any other suggestions?

Reply:
Me and Oliver synced over this today. This is what we thought - https://github.com/ossf/scorecard/issues/63 wants to address the problem that if Scorecard tool fails, the user should know "why the check failed", "how confident are we about this result" and "how to fix it". This PR won't solve that issue. A simple example - if the check failed due to too many retries, we'd simply be outputting a generic description and helpURL. This does not help our case in making the tool more user friendly.

So here's my suggestion - 

1. Let's update all of our checks to return error_codes on failure. Each unique case of failure should have a unique error_code. These can simply be appended to []details of CheckResult for now.
2. We should then update the YAML to add details/remediation corresponding to these unique failures.
3. Update cmd/root.go to include the remediation string from YAML if any of the checks fail.

What do you think? @oliverchang @inferno-chromium for feedback/opinions.

Reply:
Sounds like a plan, yes giving actual remediation instead of everything is a nice idea. and in yes in tool run, it does not make sense to give everything, (although it can say at end, for more information on check, please see doc at <link>).

Reply:
Sounds like a good plan to me too. Raising more relevant errors messages seems way more useful. @naveensrinivasan WDYT?

Reply:
SG.

Shall we also provide additional information about the checks we make in the command line output (when `--show-details` is provided)? I personally find it useful - even if somehow redundant with the YAML document file. Here's an example output (YES implicitly means it's a positive check outcome):
```
$ scorecard <> --show-details
RESULTS
-------
Repo: some-repo
Active: Fail 
	Confidence: 100%
	Commits in last 90 days greater than 2: NO (1)
Branch-Protection: Fail 
	Confidence: 60%
	AllowForcePushes disabled: YES
	AllowDeletions disabled: YES
	EnforceAdmins enabled: YES
	RequireLinearHistory enabled: YES
	RequiredStatusChecks enabled: YES but has no contexts
	RequiredPullRequestReviews enabled: YES but with only 1 reviews
	DismissStaleReviews enabled: YES
	DismissalRestrictions enabled: NO
	RequireCodeOwnerReviews enabled: YES

```

This is just a first attempt and be changed, of course.